compiler/llvm/llvmUtil.cpp

```diff
@@ -346,8 +346,18 @@ int64_t arrayVecN(llvm::Type *t)
     unsigned n = at->getNumElements();
     return n;
   } else if( t->isVectorTy() ) {
+#if HAVE_LLVM_VER >= 120
+    unsigned n;
+    if (llvm::FixedVectorType *vt = llvm::dyn_cast<llvm::FixedVectorType>(t)) {
+      n = vt->getNumElements();
+    } else {
+      assert(0 && "Scalable vector type not handled");
```

Comment:
I think this function should return `-1` for this case rather than asserting. We can update the comment to say it will return -1 for a scalable vector.